lib/WALASupport/SILWalaInstructionVisitor.cpp

```diff
@@ -389,6 +389,35 @@ jobject SILWalaInstructionVisitor::visitEndUnpairedAccessInst(EndUnpairedAccessI
   return Wala->makeNode(CAstWrapper::EMPTY);
 }
 
+jobject SILWalaInstructionVisitor::visitDeallocValueBufferInst(DeallocValueBufferInst *DVBI) {
+  SILValue BufferValue = DVBI->getOperand();
+  string ValueTypeName = DVBI->getValueType().getAsString();
+
+  if (Print) {
+    llvm::outs() << "\t [BUFFER ADDR]: " << BufferValue.getOpaqueValue() << "\n";
+    llvm::outs() << "\t [VALUE TYPE]: " << ValueTypeName << "\n";
+  }
+
+  SymbolTable.remove(BufferValue.getOpaqueValue());
+  return  Wala->makeNode(CAstWrapper::EMPTY);
+}
+
+jobject SILWalaInstructionVisitor::visitProjectValueBufferInst(ProjectValueBufferInst *PVBI) {
+  SILValue BufferValue = PVBI->getOperand();
+  string ValueTypeName = PVBI->getValueType().getAsString();
+
+  if (Print) {
+    llvm::outs() << "\t [BUFFER ADDR]: " << BufferValue.getOpaqueValue() << "\n";
+    llvm::outs() << "\t [VALUE TYPE]: " << ValueTypeName << "\n";
+  }
+
+  if (SymbolTable.has(BufferValue.getOpaqueValue())) {
```

Comment:
What if the `Buffervalue` is not in the symbol table? Shouldn't we insert it there?

Reply:
I originally thought the same but all of the ‘project’ instructions state that they are undefined behaviour if they do not exist so I don’t think so!